urllib3/util/connection.py

```diff
@@ -64,7 +66,8 @@ def create_connection(address, timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
     if host.startswith('['):
         host = host.strip('[]')
     err = None
-    for res in socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM):
+    family = supported_ip_family()
+    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
```

Comment:
Ok, so this is fine but the code is moderately unclear. I don't think `supported_ip_family` is a good name for this function: both here and lower down I was a bit confused about what it was actually achieving.

I wonder if the better approach here is to call it something like `family_filter`, and then add a comment here that explains how it works in the context of `getaddrinfo`, *and* a docstring in the actual function itself explaining why it sometimes returns zero.